crates/matrix-sdk/tests/integration/room/joined.rs

```diff
@@ -152,6 +152,32 @@ async fn unban_user() {
     room.unban_user(user, None).await.unwrap();
 }
 
+#[async_test]
+async fn mark_as_unread() {
+    let (client, server) = logged_in_client().await;
+
+    Mock::given(method("PUT"))
+        .and(path_regex(
+            r"^/_matrix/client/r0/user/.*/rooms/.*/account_data/com.famedly.marked_unread",
+        ))
+        .and(header("authorization", "Bearer 1234"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(&*test_json::EMPTY))
+        .mount(&server)
+        .await;
+
+    mock_sync(&server, &*test_json::SYNC, None).await;
+
+    let sync_settings = SyncSettings::new().timeout(Duration::from_millis(3000));
+
+    let _response = client.sync_once(sync_settings).await.unwrap();
+
+    let room = client.get_room(&DEFAULT_TEST_ROOM_ID).unwrap();
+
+    room.mark_unread(true).await.unwrap();
+
+    room.mark_unread(false).await.unwrap();
```

Comment:
I guess we would want to test `room.is_mark_unread` or something like that too? And I guess we would also want to test the content of the request? This last suggestion is probably overkill though.

Reply:
> I guess we would want to test room.is_mark_unread or something like that too?

I would need to run a sliding sync loop and I don't think this allows for that